flink-connectors/flink-connector-pubsub/src/main/java/org/apache/flink/streaming/connectors/pubsub/DefaultPubSubSubscriberFactory.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.pubsub;
```

Comment:
Pub/Sub is such a general term, that having a package like that might be a bit misleading (as if connectors in other packages are non-"publish-subscribe" ). How about `.connectors.google.pubsub` or `.connectors.gcp.pubsub` ?

Reply:
But then we need to discuss the name of the maven package and directories as well.

You are right that all other connectors have proper product names (Kafka, kinesis, elasticsearch, rabbitmq). We could consider calling the connector "flink-connector-gcp-pubsub" ? (I'm against having "google" in there :) )

What do you think @Xeli ?

Reply:
Yeah, Google likes giving their products generic names... (PubSub, Datastore, FileStore, Chat...)

I like `.connectors.gcp.pubsub` best. If we ever add other connectors for other Google Cloud platform products that'll fit well.

What should the name be on the website / docs? GCP PubSub might look a bit awkward, perhaps `Google PubSub` fits better there?

Reply:
@afedulov @rmetzger I've just added `gcp` to most packages / maven modules / artifacts.